### PR TITLE
Authentication concepts

### PR DESCRIPTION
**Pull request by psiinon**
Screenshot 2021-11-12 at 17-45-31 OWASP ZAP – Authentication - Concepts

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]

File: site/content/docs/authentication/concepts.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+These are the concepts that you will need to understand in order to configure authentication in ZAP.
+
+### Contexts
+
+ZAP [contexts](/docs/desktop/start/features/contexts/) are a way of relating a set of URLs together.
+You can define any contexts you like, but it is expected that a context will correspond to a web application.
+
+Authentication data (including the session handling and user credentials) is associated with a context, so you should have a context which represents the whole of an app protected by authentication.
```
Comment (ricekot) [inline site/content/docs/authentication/concepts.md:16]:
```suggestion
Authentication data (including the session handling and user credentials) is associated with a context, so your context should represent the entire app that is protected by authentication.
```

File: site/content/docs/authentication/concepts.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+### Sessions Handling
+
+HTTP is a stateless protocol ([RFC2616](https://www.ietf.org/rfc/rfc2616.txt) section 5), where each request and response pair is independent of other web interactions. [^1]
+
+A session ID or token binds the user authentication credentials (in the form of a user session) to the user HTTP traffic.
```
Comment (ricekot) [inline site/content/docs/authentication/concepts.md:22]: "user session" could link to `/docs/desktop/start/features/sessionmanagement/`

File: site/content/docs/authentication/concepts.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+These ID's are typically passed in the HTTP request headers but some apps have been known to pass them via URL parameters or even POST data.
```
Comment (ricekot) [inline site/content/docs/authentication/concepts.md:23]:
```suggestion
These IDs are typically passed in the HTTP request headers, but some apps have been known to pass them via URL parameters or even POST data.
```

File: site/content/docs/authentication/concepts.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+ZAP has to understand how an app manages the user session in order to set the required ID in the HTTP requests that it sends.
+ZAP has built in support for Cookie and HTTP Authentication Session Management but it can also be extended to support any session handling
+method using scripts.
+
+### Authentication Methods
+
+Authentication is the process of verifying that an individual, entity or website is whom it claims to be. Authentication in the context of web applications is commonly performed by submitting a username or ID and one or more items of private information that only a given user should know. [^2]
```
Comment (ricekot) [inline site/content/docs/authentication/concepts.md:31]:
```suggestion
Authentication is the process of verifying that an individual, entity, or website is who it claims to be. Authentication in the context of web applications is commonly performed by submitting a username or ID and one or more items of private information that only a given user should know. [^2]
```

File: site/content/docs/authentication/concepts.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+### Contexts
+
+ZAP [contexts](/docs/desktop/start/features/contexts/) are a way of relating a set of URLs together.
+You can define any contexts you like, but it is expected that a context will correspond to a web application.
+
+Authentication data (including the session handling and user credentials) is associated with a context, so you should have a context which represents the whole of an app protected by authentication.
+
+### Sessions Handling
+
+HTTP is a stateless protocol ([RFC2616](https://www.ietf.org/rfc/rfc2616.txt) section 5), where each request and response pair is independent of other web interactions. [^1]
```
Comment (thc202) [inline site/content/docs/authentication/concepts.md:20]:
```suggestion
HTTP is a stateless protocol ([RFC7230](https://datatracker.ietf.org/doc/html/rfc7230)), where each request and response pair is independent of other web interactions. [^1]
```

File: site/content/docs/authentication/concepts.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+User credentials are typically a user identifier and one or more pieces of information that only the user should know.
+
+ZAP has to have access to the user credentials in order to authenticate as that user.
+
+You can configure as many users as you need and can specify which one to use when using the ZAP spiders and active scanner.
+
+### Verification Strategies
+
+Application sessions do not typically remain open indefinitely. They can be closed after a period of time, if a logout feature is activated or in some cases if the app detects that the user is doing something inappropriate.
```
Comment (thc202) [inline site/content/docs/authentication/concepts.md:46]:
```suggestion
Application sessions do not typically remain valid indefinitely. They can be invalidated after a period of time, if a logout feature is activated or in some cases if the app detects that the user is doing something inappropriate.
```

File: site/content/docs/authentication/concepts.md (hunk 4 above)
Comment (psiinon) [reply site/content/docs/authentication/concepts.md:31]: I'd rather leave this as its a direct quote from the page linked to...

File: site/content/docs/authentication/concepts.md (hunk 5 above)
Comment (psiinon) [reply site/content/docs/authentication/concepts.md:20]: Another direct quote :/

**Comment by psiinon** [comment]
Thanks - applied all suggestions except those related to direct quotes - I'd reather leave them as is...

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**